packages/modules/vehicles/smarteq/api.py

```diff
@@ -29,27 +32,79 @@
 GUID = "280C6B55-F179-4428-88B6-E0CCF5C22A7C"
 ACCEPT_LANGUAGE = "de-de"
 
+TOKENS_REFRESH_THRESHOLD = 3600
+
+
+# helper functions
+def nested_key_exists(element: dict, *keys: str) -> bool:
+    # Check if *keys (nested) exists in `element` (dict).
+    if not isinstance(element, dict):
+        raise AttributeError('nested_key_exists() expects dict as first argument - got type ' + str(type(element)))
+    if len(keys) == 0:
+        raise AttributeError('nested_key_exists() expects at least two arguments, one given.')
+
+    _element = element
+    for key in keys:
+        try:
+            _element = _element[key]
+        except KeyError:
+            return False
+    return True
+
 
 class Api:
 
     def __init__(self, vehicle: int):
         self.log = logging.getLogger("soc."+__name__)
+        self.storeFile = str(RAMDISK_PATH) + '/soc_smarteq_store_vh_' + str(vehicle)
         # LOGLEVEL = os.environ.get('LOGLEVEL', 'INFO').upper()
         # logging.basicConfig(level=LOGLEVEL)
+
+        # self.method keeps a high level trach of actions
+        self.method = ''
+        self.soc_ts = 'n/a'
+        # self.store is read from ramdisk at start and saved at end.
+        # currently is contains:
+        # Tokens: refresh- and access-tokens of OAUTH
+        # refresh_timestamp: epoch of last refresh_tokens.
+        self.store = {}
+        self.store['Tokens'] = {}
+        self.store['refresh_timestamp'] = int(0)
+
         self.session = requests.session()
```

Comment:
```suggestion
        self.session = req.get_http_session()
```
In diesem Modul im common-Ordner sind hooks implementiert, die immer den Status abfragen, ggf loggen und immer die Antwort loggen.
Die get-Abfrage sieht wie bisher aus: `self.session.get()`